tests/cluster/tests/24-links.tcl

```diff
@@ -112,3 +108,60 @@ test "Disconnect link when send buffer limit reached" {
     $primary1 CONFIG set cluster-node-timeout $oldtimeout
     $primary1 CONFIG set cluster-link-sendbuf-limit $oldlimit
 }
+
+
+test "Link memory increases with publishes" {
+    set server [Rn 0]
+    set msg_size 10000
+    set num_msgs 10
+
+    # Publish ~100KB to one of the servers
+    $server MULTI
+    $server INFO memory
+    for {set i 0} {$i < $num_msgs} {incr i} {
+        $server PUBLISH channel [string repeat "x" $msg_size]
+    }
+    $server INFO memory
+    set res [$server EXEC]
+
+    set link_mem_before_pubs [getInfoProperty $res mem_cluster_links]
+
+    # Remove the first half of the response string which contains the
+    # first "INFO memory" results and search for the property again
+    set res [string range $res [expr [string length $res] / 2] end]
+    set link_mem_after_pubs [getInfoProperty $res mem_cluster_links]
+    
+    # We expect the memory to have increased by more than
+    # the culmulative size of the publish messages
+    set mem_diff_floor [expr $msg_size * $num_msgs]
+    set mem_diff [expr $link_mem_after_pubs - $link_mem_before_pubs]
+    assert {$mem_diff > $mem_diff_floor}
+}
+
+test "Link memory resets after publish messages flush" {
+    # Allow a little buffer to ensure publishes from the 
+    # previous test are flushed
+    after 3000
```

Comment:
This might introduce flakiness. 

I would suggest to wait for few iteration to flush the data out or fail. Or it would be good to kill the link by manipulating the config `cluster-link-sendbuf-limit`